kythe/cxx/common/path_utils.cc

```diff
@@ -71,4 +76,73 @@ std::string CleanPath(absl::string_view input) {
   return absl::StrCat(PathPrefix(input), absl::StrJoin(parts, "/"));
 }
 
+bool IsAbsolutePath(absl::string_view path) {
+  return absl::StartsWith(path, "/");
+}
+
+bool GetCurrentDirectory(std::string* dir) {
+  size_t len = 128;
+  auto buffer = absl::make_unique<char[]>(len);
+  for (;;) {
+    char* p = getcwd(buffer.get(), len);
+    if (p != nullptr) {
+      *dir = p;
+      return true;
+    } else if (errno == ERANGE) {
+      len += len;
+      buffer = absl::make_unique<char[]>(len);
+    } else {
+      return false;
+    }
+  }
+}
+
+std::string MakeCleanAbsolutePath(const std::string& in_path) {
+  std::string abs_path;
+  if (IsAbsolutePath(in_path)) {
+    abs_path = in_path;
+  } else {
+    std::string dir;
+    if (!GetCurrentDirectory(&dir)) {
+      LOG(ERROR) << "Unable to get current working directory";
+      return "";
+    }
+    abs_path = JoinPath(dir, in_path);
+  }
+  return CleanPath(abs_path);
+}
+
+std::pair<absl::string_view, absl::string_view> SplitPath(
+    absl::string_view path) {
```

Comment:
Is this conforming to an existing function? If not, prefer to use a named struct with meaningful members rather than `std::pair`.  `.first` and `.second` are pretty hard to read at callsites.

Reply:
No particular interface, I just borrowed this from google3. I added a `PathParts` struct to name the parts of the return value.